src/net/larse/lcms/algorithms/VCT.java

```diff
@@ -1570,23 +1247,27 @@ public VCTOutput(int lcType,
                      double[] distMagnVi,
                      double[] distMagnBr,
                      double[] distMagnB4) {
-      this.years = Ints.asList(Arrays.copyOfRange(years, 0, nYears));
+      this.years = Arrays.copyOfRange(years, 0, nYears);
       if (lcType != PART_FOREST) {
-        this.distFlag = new ArrayList<>(Collections.nCopies(nYears, lcType));
-        this.distMagn = new ArrayList<>(Collections.nCopies(nYears, 0.0));
-        this.distMagnVi = new ArrayList<>(Collections.nCopies(nYears, 0.0));
-        this.distMagnBr = new ArrayList<>(Collections.nCopies(nYears, 0.0));
-        this.distMagnB4 = new ArrayList<>(Collections.nCopies(nYears, 0.0));
+        this.distFlag = new int[nYears];
+        Arrays.fill(this.distFlag, lcType);
+        this.distMagn = new double[nYears];
+        Arrays.fill(this.distMagn, -1.0);
+        this.distMagnVi = new double[nYears];
+        Arrays.fill(this.distMagnVi, -1.0);
+        this.distMagnBr = new double[nYears];
+        Arrays.fill(this.distMagnBr, -1.0);
+        this.distMagnB4 = new double[nYears];
+        Arrays.fill(this.distMagnB4, -1.0);
       } else {
-        this.distFlag = Ints.asList(Arrays.copyOfRange(distFlag, 0, nYears));
-        this.distMagn = Doubles.asList(Arrays.copyOfRange(distMagn, 0, nYears));
-        this.distMagnVi = Doubles.asList(Arrays.copyOfRange(distMagnVi, 0, nYears));
-        this.distMagnBr = Doubles.asList(Arrays.copyOfRange(distMagnBr, 0, nYears));
-        this.distMagnB4 = Doubles.asList(Arrays.copyOfRange(distMagnB4, 0, nYears));
+        this.distFlag = Arrays.copyOfRange(distFlag, 0, nYears);
+        this.distMagn = Arrays.copyOfRange(distMagn, 0, nYears);
+        this.distMagnVi = Arrays.copyOfRange(distMagnVi, 0, nYears);
+        this.distMagnBr = Arrays.copyOfRange(distMagnBr, 0, nYears);
+        this.distMagnB4 = Arrays.copyOfRange(distMagnB4, 0, nYears);
```

Comment:
Yang and I talked about taking the lcType logic split out of this constructor and putting it elsewhere.  I've also coded the non-changed magnitudes to be -1.0 for testing purposes, but original code calls for them to be 0.0.